ZipBuilder/Sources/FirebaseReleaser/InitializeRelease.swift

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+import FirebaseManifest
+import Utils
+
+struct InitializeRelease {
+  static func setupRepo(gitRoot: URL) -> String {
+    let manifest = FirebaseManifest.shared
+    let branch = createReleaseBranch(path: gitRoot, version: manifest.version)
+    updatePodspecs(path: gitRoot, manifest: manifest)
+    updatePodfiles(path: gitRoot, version: manifest.version)
+    return branch
+  }
+
+  /// The branch is based on the minor version to represent this is the branch for subsequent
+  /// patches.
+  private static func createReleaseBranch(path: URL, version: String) -> String {
+    let versionParts = version.split(separator: ".")
+    let minorVersion = "\(versionParts[0]).\(versionParts[1])"
+    let branch = "release-\(minorVersion)"
+    Shell.executeCommand("git checkout master", workingDir: path)
+    Shell.executeCommand("git pull", workingDir: path)
+    Shell.executeCommand("git checkout -b \(branch)", workingDir: path)
+    return branch
+  }
+
+  /// Update the podspec versions.
+  private static func updatePodspecs(path: URL, manifest: FirebaseManifest.Manifest) {
+    for pod in manifest.pods {
+      if !pod.isClosedSource {
+        if pod.name == "Firebase" {
+          updateFirebasePodspec(path: path, manifest: manifest)
+        } else {
+          let version = pod.podVersion ??
+            (pod.isBeta ? manifest.version + "-beta" : manifest.version)
+          Shell.executeCommand("sed -i.bak -e \"s/\\(\\.version.*=[[:space:]]*'\\).*'/\\1" +
+            "\(version)'/\" \(pod.name).podspec", workingDir: path)
+        }
+      }
+    }
+  }
+
+  private static func updateFirebasePodspec(path: URL, manifest: FirebaseManifest.Manifest) {
+    let podspecFile = path.appendingPathComponent("Firebase.podspec")
+    var contents = ""
+    do {
+      contents = try String(contentsOfFile: podspecFile.path, encoding: .utf8)
+    } catch {
+      fatalError("Could not read Firebase podspec. \(error)")
+    }
+    let firebaseVersion = manifest.version
+    for firebasePod in manifest.pods {
+      if !firebasePod.isFirebase {
+        continue
+      }
+      let pod = firebasePod.name
+      let version = firebasePod.isBeta ? firebaseVersion + "-beta" : firebaseVersion
+      if pod == "Firebase" {
+        // Replace version in string like s.version = '6.9.0'
+        guard let range = contents.range(of: "s.version") else {
+          fatalError("Could not find version of Firebase pod in podspec at \(podspecFile)")
+        }
+        var versionStartIndex = contents.index(range.upperBound, offsetBy: 1)
+        while contents[versionStartIndex] != "'" {
+          versionStartIndex = contents.index(versionStartIndex, offsetBy: 1)
+        }
+        var versionEndIndex = contents.index(versionStartIndex, offsetBy: 1)
+        while contents[versionEndIndex] != "'" {
```

Comment:
Optional: `Scanner` API like [this](https://developer.apple.com/documentation/foundation/scanner/3200290-scanuptocharacters) may be a bit more expressive for such cases than enumerating the string directly.

Reply:
Added TODO.